frontend/sozisel-app/src/components/AuthRoute/AuthRoute.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { ReactElement, useEffect } from "react";
+import { Route, useHistory } from "react-router";
+
+import { useMeQuery } from "../../graphql";
+
+interface AuthRouteProps {
+  component: React.ComponentType<unknown> | undefined;
+  path: string;
+}
+
+export default function AuthRoute({
+  component,
+  path,
+}: AuthRouteProps): ReactElement {
+  const { error, loading } = useMeQuery();
+  const history = useHistory();
+
+  useEffect(() => {
+    if (error && error.message === "unauthorized") {
+      console.log(error);
+      history.push("/login");
+      localStorage.removeItem("token");
+    }
+  }, [error, history]);
+
+  return <> {!loading && <Route exact path={path} component={component} />} </>;
```

Comment:
You could change that to one `if` statement e.g. `if (loading) return <SomeLoadingComponent />` or whatever. 
And I believe we can use React Router (maybe you already use it but I haven't reached that yet), this approach looks too raw for me.